File: Lab Week 1 Part 1/dart/bin/abstraction.dart
abstract class Shape {
  double area();
}

class Circle extends Shape {
  final double r;
  Circle(this.r);

  @override
  double area() => 3.141592653589793 * r * r;
}

class Rectangle extends Shape {
  final double w, h;
  Rectangle(this.w, this.h);

  @override
  double area() => w * h;
}

void run() {
  var circle = Circle(2);
  var rectangle = Rectangle(3, 4);

  print("Circle area = ${circle.area()}");
  print("Rectangle area = ${rectangle.area()}");
}

File: Lab Week 1 Part 1/java/Abstraction.java
public class Abstraction {
    static abstract class Shape {
        abstract double area();
    }

    static class Circle extends Shape {
        private double r;
        public Circle(double r) { this.r = r; }
        @Override
        double area() { return Math.PI * r * r; }
    }

    static class Rectangle extends Shape {
        private double w, h;
        public Rectangle(double w, double h) { this.w = w; this.h = h; }
        @Override
        double area() { return w * h; }
    }

    public static void run() {
        Shape c = new Circle(2);
        Shape r = new Rectangle(3, 4);
        System.out.println("Circle area = " + c.area());
        System.out.println("Rectangle area = " + r.area());
    }
}
